File: app/backend/feed/src/main/java/org/petify/feed/model/EventParticipant.java
package org.petify.feed.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.apache.commons.lang.builder.EqualsBuilder;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "event_participants")
public class EventParticipant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "event_id", nullable = false)
    private Long eventId;

    @Column(name = "username", nullable = false)
    private String username;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }

        if (!(o instanceof EventParticipant participant)) {
            return false;
        }

        return new EqualsBuilder()
                .append(eventId, participant.eventId)
                .append(username, participant.username)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new org.apache.commons.lang.builder.HashCodeBuilder(17, 37)
                .append(eventId)
                .append(username)
                .toHashCode();
    }

    @Override
    public String toString() {
        return new org.apache.commons.lang.builder.ToStringBuilder(this)
                .append("id", id)
                .append("eventId", eventId)
                .append("username", username)
                .toString();
    }
}

File: app/mobile/lib/models/event_participant.dart
class EventParticipant {
  final int id;
  final int eventId;
  final String username;
  final DateTime createdAt;

  EventParticipant({
    required this.id,
    required this.eventId,
    required this.username,
    required this.createdAt,
  });

  factory EventParticipant.fromJson(Map<String, dynamic> json) {
    int _parse(dynamic v) => v is int ? v : int.parse(v.toString());
    return EventParticipant(
      id: _parse(json['id']),
      eventId: _parse(json['eventId']),
      username: json['username'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}